src/bin/vhost_user_fs.rs

```diff
@@ -70,7 +71,7 @@ impl convert::From<Error> for io::Error {
 }
 
 struct VhostUserFsBackend<F: FileSystem + Send + Sync + 'static> {
-    mem: Option<GuestMemoryMmap>,
+    mem: Option<Arc<ArcSwap<GuestMemoryMmap>>>,
```

Comment:
I don't understand why this change is necessary?

Reply:
The memory never gets modified (maybe we should do that, @sboeuf - if we have a hotplug do we need to tell the vhost-user backends about a different memory map?) so you don't need to protect it with an Arc<Mutex<T>> or a Arc<ArcSwap<T>> just an Arc<T> should be viable.